Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -610,4 +627,114 @@ namespace AZ::Render
+        }
+    }
+
+    void AtomActorDebugDraw::RenderLineAxis(
+        AzFramework::DebugDisplayRequests* debugDisplay,
+        AZ::Transform worldTM,
+        float size,
+        bool selected,
+        bool renderAxis,
+        bool renderAxisName)
+    {
+        const float axisHeight = size * 0.7f;
+        const float frontSize = size * 5.0f + 0.2f;
+        const AZ::Vector3 position = worldTM.GetTranslation();
+
+        if (renderAxis)
+        {
+            AZ::Color xSelectedColor;
+            if (selected)
+            {
+                xSelectedColor = AZ::Colors::Orange;
+            }
+            else
+            {
+                xSelectedColor = AZ::Colors::Red;
+            }
+
+            const AZ::Vector3 xAxisDir = (worldTM.TransformPoint(AZ::Vector3(size, 0.0f, 0.0f)) - position).GetNormalized();
+            const AZ::Vector3 xAxisArrowStart = position + xAxisDir * axisHeight;
+            debugDisplay->SetColor(xSelectedColor);
+            debugDisplay->DrawArrow(position, xAxisArrowStart, size);
+
+            if (renderAxisName)
+            {
+                const AZ::Vector3 xNamePos = position + xAxisDir * (size * 1.15f);
+                debugDisplay->DrawTextLabel(xNamePos, frontSize, "X");
+            }
+        }
+
+        if (renderAxis)
+        {
+            AZ::Color ySelectedColor;
+            if (selected)
+            {
+                ySelectedColor = AZ::Colors::Orange;
+            }
+            else
+            {
+                ySelectedColor = AZ::Colors::Blue;
+            }
+
+            const AZ::Vector3 yAxisDir = (worldTM.TransformPoint(AZ::Vector3(0.0f, size, 0.0f)) - position).GetNormalized();
+            const AZ::Vector3 yAxisArrowStart = position + yAxisDir * axisHeight;
+            debugDisplay->SetColor(ySelectedColor);
+            debugDisplay->DrawArrow(position, yAxisArrowStart, size);
+
+            if (renderAxisName)
+            {
+                const AZ::Vector3 yNamePos = position + yAxisDir * (size * 1.15f);
+                debugDisplay->DrawTextLabel(yNamePos, frontSize, "Y");
+            }
+        }
+
+        if (renderAxis)
+        {
+            AZ::Color zSelectedColor;
+            if (selected)
+            {
+                zSelectedColor = AZ::Colors::Orange;
+            }
+            else
+            {
+                zSelectedColor = AZ::Colors::Green;
+            }
+
+            const AZ::Vector3 zAxisDir = (worldTM.TransformPoint(AZ::Vector3(0.0f, 0.0f, size)) - position).GetNormalized();
+            const AZ::Vector3 zAxisArrowStart = position + zAxisDir * axisHeight;
+            debugDisplay->SetColor(zSelectedColor);
+            debugDisplay->DrawArrow(position, zAxisArrowStart, size);
+
+            if (renderAxisName)
+            {
+                const AZ::Vector3 zNamePos = position + zAxisDir * (size * 1.15f);
+                debugDisplay->DrawTextLabel(zNamePos, frontSize, "Z");
+            }
+        }
```

Comment:
suggestion: These three blocks all look very similar 😛 It might be worth factoring this code out into a function and calling it three times (up to you but just a thought)